mathjax3-ts/output/chtml/Wrappers/msqrt.ts

```diff
@@ -0,0 +1,193 @@
+        surd.getStretchedVariant([this.surdH, 0]);
+    }
+
+    /*
+     * Create an mo wrapper with the given text;
+     *
+     * @param{string} text  The text for the wrapped element
+     * @return{CHTMLWrapper}  The wrapped MmlMo node
+     */
+    protected createMo(text: string) {
+        const mmlFactory = (this.node as AbstractMmlNode).factory;
+        const node = this.wrap(mmlFactory.create('mo', {}, [(mmlFactory.create('text') as TextNode).setText(text)])) as CHTMLmo;
+        node.parent = this;
+        return node;
+    }
+
+    /*
+     * @override
+     */
+    public toCHTML(parent: HTMLElement) {
+        const surd = this.childNodes[this.surd];
+        const base = this.childNodes[this.base];
+        //
+        //  Get the parameters for the spacing of the parts
+        //
+        const sbox = surd.getBBox();
+        const bbox = base.getBBox();
+        const [p, q] = this.getPQ(sbox);
+        const [x] = this.getRootDimens(sbox);
+        const t = this.font.params.rule_thickness;
+        const T = this.font.params.surd_height;
+        //
+        //  Create the HTML structure for the root
+        //
+        const CHTML = this.standardCHTMLnode(parent);
+        let SURD, BASE, ROOT, root;
+        if (this.root != null) {
+            ROOT = CHTML.appendChild(this.html('mjx-root'));
+            root = this.childNodes[this.root];
+        }
+        const SQRT = CHTML.appendChild(this.html('mjx-box', {
+            style: {paddingTop: this.px(2 * t - T, 1)}
+        }, [
+            SURD = this.html('mjx-surd'),
+            BASE = this.html('mjx-box', {style: {
+                paddingTop: this.px(q, 1),
+                borderTop: this.px(T * bbox.scale, 1) + ' solid'
+            }})
+        ]));
+        //
+        //  Add the child content
+        //
+        this.addRoot(ROOT, root, sbox);
+        surd.toCHTML(SURD);
+        base.toCHTML(BASE);
+    }
+
+    /*
+     * Add root HTML (overridden in mroot)
+     *
+     * @param{HTMLElement} ROOT   The container for the root
+     * @param{CHTMLWrapper} root  The wreapped MML root content
+     * @param{BBox} sbox          The bounding box of the surd
+     */
+    protected addRoot(ROOT: HTMLElement, root: CHTMLWrapper, sbox: BBox) {
+    }
+
+    /*
+     * @override
+     */
+    public computeBBox() {
+        const BBOX = BBox.empty();
+        const sbox = this.childNodes[this.surd].getBBox();
+        const bbox = new BBox(this.childNodes[this.base].getBBox());
+        const [p, q] = this.getPQ(sbox);
+        const [x] = this.getRootDimens(sbox);
+        const t = this.font.params.rule_thickness;
+        const H = bbox.h + q + t;
+        bbox.h += q + 2 * t;  // FIXME:  should take into accound minimums for this.px() used above
+        this.combineRootBBox(BBOX, sbox);
+        BBOX.combine(sbox, x, H - sbox.h);
+        BBOX.combine(bbox, x + sbox.w, 0);
+        BBOX.clean();
+        return BBOX;
```

Comment:
I find bbox vs BBOX very confusing.

Reply:
If one can't use different capitalizations, why does the language have case-sensitive names.  :-)

OK, I've changed `BBOX` to just `box`.  This will probably cause merge conflicts in a later PR, since I have modified this function once I had the actual display working and found out that the layout needed work.